### DISCUSSION TITLE
Run preinstall / postinstall scripts on single package installation

### DISCUSSION THREAD

**Discussion by karlhorky** [npm/feedback#217] [CLI] [answered]
Hi there, first of all, thanks for all of the hard work on npm and everything surrounding it! Really cool what you are doing.

The [behavior of `npm`](https://npm.community/t/preinstall-npm-hook-doesnt-execute-when-installing-a-specific-package/2505) to not run `preinstall` or `postinstall` scripts specified by the developer in the project on every `npm install <pkg name>` is confusing and leads to support requests such as https://github.com/npm/cli/issues/1732

Would you consider starting to run these scripts once during installation of a package?

**Edit:** also opened on the RFC repo: https://github.com/npm/rfcs/issues/325

**Comment by MylesBorins** [comment, marked as answer]
@karlhorky I think if you want to propose a specific change to the CLI the best approach would be to start with an issue over at the [CLI RFC repo](https://github.com/npm/rfcs)

It is worth noting that a change like this could be considered breaking, which could be one of the reasons for not instituting it, or rolling it out slowly.

**Reply by karlhorky** [reply]
Ok, thanks for the direction :) Opened an issue on the RFC repo: https://github.com/npm/rfcs/issues/325
